api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java

```diff
@@ -137,7 +137,13 @@ private static ConfigProviderResolver loadSpi(ClassLoader cl) {
         }
 
         // start from the root CL and go back down to the TCCL
-        ConfigProviderResolver instance = loadSpi(cl.getParent());
+        ClassLoader parentcl = AccessController.doPrivileged(new PrivilegedAction<ClassLoader>() {
+            @Override
+            public ClassLoader run() {
+                return cl.getParent();
+            }
+        });
+        ConfigProviderResolver instance = loadSpi(parentcl);
```

Comment:
Can you put this line inside doPrivileged call? Otherwise, parentcl might be null.

Reply:
parentcl is allowed to be null (it will be when you reach the root) and there is a null check at the top of loadSpi ... functionally that hasn't changed by adding the doPriv

Reply:
This is not expected behavior though. What we want is that instance should be set to null if something is wrong in the run method though.

Reply:
double checked the api. The condition I worried about should not occur silently.